### PR TITLE
FEATURE: add support for Vulnerability Disclosures

### PR DESCRIPTION
**Pull request by madpah**

### CODE REVIEW

**Comment by stevespringett** [comment]
FYI, this will likely change in v1.4 of the spec, due in part to the inclusion of vulnerability information into the core spec, rather than an extension. Might want to keep that in mind for design purposes.

**Comment by madpah** [comment]
> FYI, this will likely change in v1.4 of the spec, due in part to the inclusion of vulnerability information into the core spec, rather than an extension. Might want to keep that in mind for design purposes.

Thanks @stevespringett - is there an approximate ETA for v1.4 being published out of interest?

File: cyclonedx/output/xml.py [hunk 1]
```diff
@@ -30,24 +31,49 @@ class Xml(BaseOutput, BaseSchemaVersion):
...
         bom = self._get_bom_root_element()
 
         if self.bom_supports_metadata():
             bom = self._add_metadata(bom=bom)
 
+        if self.get_bom().has_vulnerabilities():
+            ElementTree.register_namespace('v', Xml.get_vulnerabilities_namespace())
+
         components = ElementTree.SubElement(bom, 'components')
+        # if self.get_bom().has_vulnerabilities():
```
Comment (jkowalleck) [inline cyclonedx/output/xml.py:48]: purpose of this comment?

**Review by jkowalleck — COMMENTED** [review]
dont see any issues with the imeplementations.
could you add an example file for the JSON outcode and have it parsed in  test? 
the used schema file for JSON is in the repo already, afaik.

File: cyclonedx/output/xml.py (hunk 1 above)
Comment (madpah) [reply cyclonedx/output/xml.py:48]: Good catch @jkowalleck  - should not be there - will fix.

**Comment by madpah** [comment]
> dont see any issues with the imeplementations.
> could you add an example file for the JSON outcode and have it parsed in test?
> the used schema file for JSON is in the replo already, afaik.

My understanding is that Vulnerability extension is not supported in JSON, hence I've not added anything for it. Am I wrong?

**Comment by jkowalleck** [comment]
re: https://github.com/CycloneDX/cyclonedx-python-lib/pull/5#issuecomment-919942068
that's what it stays on https://cyclonedx.org/use-cases/#vulnerability-disclosure: 
> The vulnerability schema extension is only available in XML. It is not currently available in JSON.

so i would think you are right.
but on the other hand there is a JSON schema for the vulns: 
* https://github.com/CycloneDX/cyclonedx-python-lib/blob/main/cyclonedx/schema/ext/vulnerability-1.0-SNAPSHOT.schema.json
* https://github.com/CycloneDX/specification/blob/master/schema/ext/vulnerability-1.0-SNAPSHOT.schema.json

So it looks like vuln could be available for the JSON schema, too.
@stevespringett could you clarify here ?

**Comment by madpah** [comment]
And this is of course only for schema versions 1.1 through 1.3…

**Pull request merged**